### PR TITLE
Use custom user model

### PR DESCRIPTION
**Pull request by 2e12**
The predefined user model was not as flexible as expected. Therefore a custom model is implemented.

Closes https://github.com/2e12/306-2gether/issues/31

### CODE REVIEW

File: backend/users/tags.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+def get_tag_by_name(db, name):
+    return db.query(Tag).filter_by(name=name).first()
+
+
+def get_tags(db: Session, skip: int, limit: int):
+    return db.query(Tag).order_by(Tag.name.asc()).offset(skip).limit(limit).all()
+
+
+def resolve_tags(db: Session, tags: List[TagBaseSchema]) -> List[TagSchema]:
```
Comment (Ha21iko) [inline backend/users/tags.py:29]: Will the front-end manage other validation aspects such as the max length of the tag and other things that are not valid?

Comment (Ha21iko) [reply backend/users/tags.py:29]: I like the approach of other functions like is_user_valid() -> backend/users/useractions.py

**Review by Ha21iko — APPROVED** [review]
LGTM

File: backend/users/tags.py (hunk 1 above)
Comment (2e12) [reply backend/users/tags.py:29]: Usually you would define a validator in the schemas (https://pydantic-docs.helpmanual.io/usage/validators/) or directly in the model with string length. I this case a string length in the model should be totally fine. I will add to it in the coming days.

**Pull request merged**